### PR TITLE
Add support for Required Workflows

### LINKED ISSUE #2978: Support for Required Workflows in rulesets
GitHub has recently released https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-workflows-to-pass-before-merging. It would be nice to have support for this feature. 

References:

[Ruleset API docs](https://docs.github.com/en/rest/orgs/rules?apiVersion=2022-11-28#create-an-organization-repository-ruleset)
[Required workflows is being removed from the API](https://github.com/google/go-github/issues/2907) #2907

### PR DESCRIPTION
**Pull request by claystation**
Fixes #2978  
Suggestion to add support for Required Workflows in Github API.

### CODE REVIEW

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:105]:
```suggestion
	Ref          *string `json:"ref,omitempty"`
	RepositoryID *int64 `json:"repository_id,omitempty"`
	Sha          *string `json:"sha,omitempty"`
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:110]:
```suggestion
	RequiredWorkflows []*RuleRequiredWorkflow `json:"workflows"`
```

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:355]:
```suggestion
// NewRequiredWorkflowsRule creates a rule to require which status checks must pass before branches can be merged into a branch rule.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @claystation !
After a few tweaks, we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by claystation** [comment]
Hey @gmlewis ! Thanks for the review :) I updated the code to reflect the correct suggestions!

File: github/repos_rules.go
Comment (gmlewis) [inline github/repos_rules.go:105]: Please add the pointers to the `omitempty` fields.

```suggestion
	Ref          *string `json:"ref,omitempty"`
	RepositoryID *int64 `json:"repository_id,omitempty"`
	Sha          *string `json:"sha,omitempty"`
```

**Comment by gmlewis** [comment]
It looks like some `gofmt` needs to be called, then we should be good to move forward on this PR.

**Comment by claystation** [comment]
Hey @gmlewis! I did a `gofmt` and generated files as requested in the check :) Should be good to go now!

**Review by gmlewis — APPROVED** [review]
Thank you, @claystation !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
